fedmsg/crypto/utils.py

```diff
@@ -98,56 +99,71 @@ def validate_policy(topic, signer, routing_policy, nitpicky=False):
+            is deprecated and will be removed in a future release.
+        ca_location (str): The location of the Certificate Revocation List. This should
+            be the absolute path to a PEM-encoded file. It can also be an HTTPS url, but
+            this is deprecated and will be removed in a future release.
+        invalidate_cache (bool): Whether or not to invalidate the certificate cache.
+
+    Returns:
+        tuple: A tuple of the (CA certificate, CRL) as unicode strings.
 
+    Raises:
+        requests.exception.RequestException: Any exception requests could raise.
+        IOError: If the location provided could not be opened and read.
     """
-    alternative_cache = os.path.expanduser("~/.local" + cache)
+    if crl_location is None:
+        crl_location = ''
 
     try:
-        modtime = os.stat(cache).st_mtime
-    except OSError:
-        # File does not exist yet.
-        try:
-            # Try alternative location.
-            modtime = os.stat(alternative_cache).st_mtime
-            # It worked!  Use the alternative location
-            cache = alternative_cache
-        except OSError:
-            # Neither file exists
-            modtime = 0
-
-    if (
-        (not modtime and not cache_expiry) or
-        (cache_expiry and time.time() - modtime > cache_expiry)
-    ):
-        try:
-            with requests.Session() as session:
-                session.mount('http://', requests.adapters.HTTPAdapter(max_retries=tries))
-                session.mount('https://', requests.adapters.HTTPAdapter(max_retries=tries))
-                response = session.get(location, timeout=30)
-            with open(cache, 'w') as f:
-                f.write(response.text)
-        except IOError:
-            # If we couldn't write to the specified cache location, try a
-            # similar place but inside our home directory instead.
-            cache = alternative_cache
-            usr_dir = '/'.join(cache.split('/')[:-1])
-
-            if not os.path.isdir(usr_dir):
-                os.makedirs(usr_dir)
-
-            with open(cache, 'w') as f:
-                f.write(response.text)
-
-    return cache
+        if invalidate_cache:
+            del _cached_certificates[ca_location + crl_location]
+        else:
+            return _cached_certificates[ca_location + crl_location]
+    except KeyError:
+        pass
+
+    ca, crl = None, None
+    if ca_location:
+        ca = _load_certificate(ca_location)
+    if crl_location:
+        crl = _load_certificate(crl_location)
+
+    _cached_certificates[ca_location + crl_location] = ca, crl
+    return ca, crl
+
+
+def _load_certificate(location):
+    """
+    Load a certificate from the given location.
+
+    Args:
+        location (str): The location to load. This can either be an HTTPS URL or an absolute file
+            path. This is intended to be used with PEM-encoded certificates and therefore assumes
+            ASCII encoding.
```

Comment:
Might be good to mention the deprecation here too.

Reply:
I made a note of it in the public interface